src/Storages/Distributed/DirectoryMonitor.cpp

```diff
@@ -131,6 +132,9 @@ void StorageDistributedDirectoryMonitor::run()
 {
     std::unique_lock lock{mutex};
 
+    /// This metric will be updated with the number of pending files later.
+    CurrentMetrics::Increment metric_pending_files{CurrentMetrics::DistributedFilesToInsert, 0};
```

Comment:
So now it is possible to get 0 instead of real values until processFiles will update this metric?

Reply:
It can be "false" zero value only at startup time of DirectoryMonitor.
Then it will correspond to the number of files that we discovered.
The value can be slightly behind if new files have just appeared but we are not aware yet.

Reply:
Got it, did not look carefully into the `Increment` ctor